sync/main.go

```diff
@@ -0,0 +1,210 @@
+		ActivationTime time.Time   `json:"activationTime"`
+		FromPool       interface{} `json:"fromPool"`
+	} `json:"meta"`
+	Payload struct {
+		ID   int `json:"id"`
+		User struct {
+			ID                   int    `json:"id"`
+			Username             string `json:"username"`
+			FirstName            string `json:"firstName"`
+			LastName             string `json:"lastName"`
+			FullName             string `json:"fullName"`
+			Gender               string `json:"gender"`
+			ProfilePicture       string `json:"profilePicture"`
+			InternalEmailAddress string `json:"internalEmailAddress"`
+		} `json:"user"`
+		Pool   interface{} `json:"pool"`
+		Status string      `json:"status"`
+	} `json:"payload"`
+}
+
+func main() {
+	flag.Parse()
+
+	if len(*wsUrl) == 0 {
+		log.Fatal("Missing wsUrl")
+	}
+	if *eventId == 0 {
+		log.Fatal("Missing eventId")
+	}
+
+	log.SetFlags(0)
+
+	interrupt := make(chan os.Signal, 1)
+	signal.Notify(interrupt, os.Interrupt)
+
+	u, _ := url.Parse(*wsUrl)
+	jwt := u.Query()["jwt"][0]
+	log.Printf("connecting to %s", u.String())
+
+	c, _, err := websocket.DefaultDialer.Dial(u.String(), nil)
+	if err != nil {
+		log.Fatal("dial:", err)
+	}
+	defer c.Close()
+
+	done := make(chan struct{})
+
+	go func() {
+		defer close(done)
+		for {
+			_, message, err := c.ReadMessage()
+			if err != nil {
+				log.Println("read:", err)
+				return
+			}
+			var abstractAction LegoAction
+			err = json.Unmarshal(message, &abstractAction)
+			if err != nil {
+				log.Printf("Error decding msg: %q with err: %e\n", message, err)
+				continue
+			}
+			switch abstractAction.Type {
+			case "Event.SOCKET_REGISTRATION.SUCCESS":
+				var action LegoRegisterAction
+				_ = json.Unmarshal(message, &action)
+
+				if action.Meta.EventID != *eventId {
+					break
+				}
+
+				log.Printf("Registering user %q in VOTE\n", action.Payload.User.FullName)
+				client := &http.Client{}
+
+				legoGet, _ := http.NewRequest("GET", *legoUserEndpont+action.Payload.User.Username, nil)
+				legoGet.Header.Add("Authorization", "JWT "+jwt)
+
+				resp, err := client.Do(legoGet)
+				if err != nil {
+					log.Printf("Error when fetching lego user: %e\n", err)
+					break
+				}
+				var voteFormData struct {
+					Email string `json:"email"`
+				}
+				defer resp.Body.Close()
+				respData, _ := ioutil.ReadAll(resp.Body)
+
+				log.Printf("Got this from LEGO: %q\n", respData)
+				err = json.Unmarshal(respData, &voteFormData)
+
+				if err != nil {
+					log.Printf("Error when fetching lego user: %e\n", err)
+					break
+				}
+
+				out, _ := json.Marshal(voteFormData)
+				log.Printf("Posting to VOTE: %q", string(out))
+				req, _ := http.NewRequest("POST", *voteEndpoint, bytes.NewBuffer(out))
+				req.Header.Add("CSRF-Token", *csrfToken)
+				req.Header.Add("content-type", "application/json;charset=UTF-8")
+				req.Header.Set("Cookie", *authCookieVote)
+
+				resp, err = client.Do(req)
+
+				defer resp.Body.Close()
+				respData, _ = ioutil.ReadAll(resp.Body)
```

Comment:
Could this give an error?

Reply:
Yes and no. We don't handle 100% of errors in lego in general, but will fix error handling when stuff works as expected